app/models/submission.rb

```diff
@@ -86,6 +86,18 @@ def viewable_by?(user)
     iteration&.published?
   end
 
+  def solution_files
+    files.each_with_object(solution.exercise_solution_files) do |file, files|
+      type = files.key?(file.filename) ? :solution : :legacy
+
+      files[file.filename] = {
+        type: type,
+        content: file.content,
+        digest: file.digest
+      }
+    end
+  end
```

Comment:
This is quite confusing because we have an external `files` method and the same named (but different) internal `files` block param.

Also I found the intersection of the two arrays in the `each_with_object` quite confusing to reason about. But I think maybe just a comment that explains that is best.

```suggestion
    # Merge the submission files into the exercise files. If we find a 
    # file we don't expect, that it as type: :legacy
    files.each_with_object(solution.exercise_solution_files) do |file, merged_files|
      type = merged_files.key?(file.filename) ? :solution : :legacy

      merged_files[file.filename] = {
        type: type,
        content: file.content,
        digest: file.digest
      }
    end
  end
```